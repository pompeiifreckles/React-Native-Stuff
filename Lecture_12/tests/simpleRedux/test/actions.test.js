/* eslint-disable no-unused-vars */
/* global jest, expect, describe, it */

const { addUser, addContact, contactFetch } = require('../index')

describe('Actions: addUser, contactFetch', () => {
  it('addUser [Function]: (newUser)', () => {
    expect(addUser('Donald')).toEqual({ type: 'UPDATE_USER', payload: 'Donald' })
  })

  it('addContact [Function]: (newContact)', () => {
    expect(addContact({ name: 'John Doe', phone: '9999888877' })).toMatchSnapshot()
  })

  it('contactFetch [Closure-Function]: () => async(dispatch)', async () => {
    const mockDispatch = jest.fn()
    await contactFetch()(mockDispatch)

    // console.log(mockDispatch.mock.calls[1][0])
    expect(mockDispatch.mock.calls[0][0]).toEqual({ type: 'FETCHING_CONTACT', payload: null })
    expect(mockDispatch.mock.calls[1][0]).toMatchSnapshot({ payload: { name: expect.any(String), phone: expect.any(String) } })
  })
})