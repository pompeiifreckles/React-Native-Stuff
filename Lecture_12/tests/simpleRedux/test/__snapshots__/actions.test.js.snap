// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions: addUser, contactFetch addContact [Function]: (newContact) 1`] = `
Object {
  "payload": Object {
    "name": "John Doe",
    "phone": "9999888877",
  },
  "type": "UPDATE_CONTACT",
}
`;

exports[`Actions: addUser, contactFetch contactFetch [Closure-Function]: () => async(dispatch) 1`] = `
Object {
  "payload": Object {
    "name": Any<String>,
    "phone": Any<String>,
  },
  "type": "FETCHED_CONTACT",
}
`;
